peripherals/eos-evm-ws-proxy/block-monitor.js

```diff
@@ -130,9 +130,14 @@ class BlockMonitor extends EventEmitter {
         }
       }
 
-      if( found_next_block == true ) {
+      while(this.reversible_blocks.length > 1024) { // protection of memory grow in case leap rpc not working
+        this.remove_front_block();
+      }
+
+      if( found_next_block == true && this.reversible_blocks.length > 180 + 30) { // reduce frequency of calling get_evm_lib
         const evm_lib = await this.get_evm_lib();
-        while(this.reversible_blocks.length > 0 && this.reversible_blocks.peek().number < evm_lib) {
+        // keep at least 180 blocks in case evm-node is out of sync with leap
+        while(this.reversible_blocks.length > 180 && this.reversible_blocks.peek().number < evm_lib) {
           this.remove_front_block();
```

Comment:
In case we are on a minority fork, we will be having different behaviors from the client perspective based on how long the fork takes to solve (since we will be removing reversible blocks from our queue)
Are we ok with this?